tools/module_dependencies.py

```diff
@@ -0,0 +1,149 @@
+__all__ = ['modules_dependent_on', 'dependency_graph']
+
+
+import numpy as np
+import importlib
+import multiprocessing
+import functools
+import sys
+import os
+import glob
+import pytest
+
+
+package = 'skimage'
+included_mods = [
+    '_shared',
+]
+excluded_mods = [
+    '__version__',
+]
+
+
+@functools.cache
+def _pkg_modules() -> tuple[list[str], dict[str, int]]:
+    """List all package submodules.
+
+    Returns
+    -------
+    submodules : list
+        Sorted list of package submodules.
+    submodule_idx : dict
+        Mapping of submodule to integer, its index
+        in `submodules`.
+    """
+    pkg = importlib.import_module(package)
+
+    members = {f'{package}.{attr}' for attr in dir(pkg)}
+    included_mods_full = {f'{package}.{mod}' for mod in included_mods}
+    excluded_mods_full = {f'{package}.{mod}' for mod in excluded_mods}
+
+    submodules = sorted((members | included_mods_full) - excluded_mods_full)
+
+    # Sort entries, so that all adjacency matrix calculations are stable
+    submodule_idx = {mod: index for index, mod in enumerate(submodules)}
+
+    # # Clear out sys.modules for when we spawn our import detector
```

Comment:
Did you find out that there was no way to spawn completely independent processes, or should I look into it?

Reply:
I think the choice of how exactly they spawn is platform-dependent, so not sure how much control we have using `subprocessing`. We could use a different mechanism, but probably not necessary unless we discover a problem with the current approach.